source/common/ssl/ssl_socket.cc

```diff
@@ -386,27 +386,55 @@ ClientSslSocketFactory::ClientSslSocketFactory(ClientContextConfigPtr config,
                                                Ssl::ContextManager& manager,
                                                Stats::Scope& stats_scope)
     : manager_(manager), stats_scope_(stats_scope), config_(std::move(config)),
-      ssl_ctx_(manager_.createSslClientContext(stats_scope_, *config_)) {}
+      ssl_ctx_(manager_.createSslClientContext(stats_scope_, *config_)) {
+  if (config_->getDynamicSecretProvider()) {
+    config_->getDynamicSecretProvider()->addUpdateCallback(*this);
+  }
+}
+
+ClientSslSocketFactory::~ClientSslSocketFactory() {
+  if (config_->getDynamicSecretProvider()) {
+    config_->getDynamicSecretProvider()->removeUpdateCallback(*this);
+  }
+}
 
 Network::TransportSocketPtr ClientSslSocketFactory::createTransportSocket() const {
   return std::make_unique<Ssl::SslSocket>(ssl_ctx_, Ssl::InitialState::Client);
```

Comment:
Need to check ssl_ctx_ is null.  Same as ServerSslSocketFactory.

Reply:
Added nullptr check, and return nullptr if ssl_ctx_ is nullptr. I think we have covered the case when transport socket is nullptr in ClusterManagerImpl::tcpConnForCluster, and ConnectionHandlerImpl::ActiveListener::newConnection